sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -201,6 +201,67 @@ mod checked {
         (inner, effects, execution_result)
     }
 
+    #[instrument(
+        name = "commit_transaction_without_execution",
+        level = "debug",
+        skip_all
+    )]
+    pub fn commit_transaction_without_execution<Mode: ExecutionMode>(
+        shared_object_refs: Vec<ObjectRef>,
+        mut temporary_store: TemporaryStore<'_>,
+        transaction_kind: TransactionKind,
+        transaction_signer: SuiAddress,
+        gas_charger: &mut GasCharger,
+        transaction_digest: TransactionDigest,
+        mut transaction_dependencies: BTreeSet<TransactionDigest>,
+        epoch_id: &EpochId,
+        enable_expensive_checks: bool,
+    ) -> (
+        InnerTemporaryStore,
+        TransactionEffects,
+        Result<Mode::ExecutionResults, ExecutionError>,
+    ) {
+        let error = ExecutionError::new(ExecutionErrorKind::CertificateDenied, None);
```

Comment:
I don't think we should overload this error code for shared object deletion - it is part of the public API seen by users and the error message will be misleading. You've already added a new ExecutionErrorKind in this PR, lets add one more for `SharedObjectDeleted`